### PR TITLE
Try to get hold of URL and instance name for disambiguation

### PR DESCRIPTION
**Pull request by danimo**
fixes https://github.com/ChristophWurst/twofactor_totp/issues/3

### CODE REVIEW

File: lib/Controller/SettingsController.php [hunk 1]
```diff
@@ -68,9 +68,17 @@ public function enable($state) {
...
 			$secret = $this->totp->createSecret($user);
-			
 			$qrCode = new QrCode();
-			$qr = $qrCode->setText("otpauth://totp/ownCloud%20TOTP?secret=$secret")
+			$nameLine = $user->getDisplayName();
+			$loginName = \OC::$server->getUserSession()->getSession()->get('loginname');
+			if (!empty($loginName)) {
+				$nameLine = "$nameLine ($loginName)";
+			}
+			$encodedNameLine = rawurlencode(nameLine);
```
Comment (LukasReschke) [inline lib/Controller/SettingsController.php]: $nameLine

Comment (danimo) [inline lib/Controller/SettingsController.php]: Fixed

File: lib/Controller/SettingsController.php [hunk 2]
```diff
@@ -68,9 +68,17 @@ public function enable($state) {
 		$user = $this->userSession->getUser();
 		if ($state) {
 			$secret = $this->totp->createSecret($user);
-			
 			$qrCode = new QrCode();
-			$qr = $qrCode->setText("otpauth://totp/ownCloud%20TOTP?secret=$secret")
+			$nameLine = $user->getDisplayName();
+			$loginName = \OC::$server->getUserSession()->getSession()->get('loginname');
```
Comment (ChristophWurst) [inline lib/Controller/SettingsController.php]: either use ``$this->userSession`` or inject ``ISession``

File: lib/Controller/SettingsController.php [hunk 3]
```diff
@@ -68,9 +68,17 @@ public function enable($state) {
...
+			if (!empty($loginName)) {
+				$nameLine = "$nameLine ($loginName)";
+			}
+			$encodedNameLine = rawurlencode($nameLine);
+			$instanceRoot = \OC::$server->getUrlGenerator()->getAbsoluteURL('/');
```
Comment (ChristophWurst) [inline lib/Controller/SettingsController.php]: inject ``IURLGenerator`` via the constructor

File: lib/Controller/SettingsController.php [hunk 4]
```diff
@@ -68,9 +68,17 @@ public function enable($state) {
...
+			$productName = (new \OCP\Defaults())->getName();
```
Comment (ChristophWurst) [inline lib/Controller/SettingsController.php]: inject if possible

File: lib/Controller/SettingsController.php (hunk 2 above)
Comment (ChristophWurst) [inline lib/Controller/SettingsController.php]: is 'loginname' always set? This looks error-prone, an app developer wouldn't know that. Maybe we should think about a proper public API method to get the login name?

**Comment by ChristophWurst** [comment]
Superseded by #35. Thanks :-)

**Pull request closed without merging**